### PR TITLE
 Update dataanalysis api readme with new approach

### PR DESCRIPTION
**Pull request by seethalelias-aot**
# Issue Tracking

JIRA: 
Issue Type: BUG/ FEATURE

# Changes

# How has the change been tested?

# Screenshots (if applicable)

# Build Success screenshot (Till a CICD pipeline is set up)

# Notes

### CODE REVIEW

File: forms-flow-data-analysis-api/README.md [hunk 1]
```diff
@@ -6,43 +6,32 @@
...
-from public sources like customer reviews of BC Government services centres from google maps.
+So we have collected the initial data from a generic dataset in Kaggle. Dataset is a combined Mobile reviews, Twitter sentiment, Yelp review, Toxic reviews and few more to cover multiple domain of sentiment analysis.
 
-2. The input dataset was cleaned using various preprocessing techniques and transformed into tidy text data.
- 
-3. The collected Data was annotated using a NER analyser tool like Docanno to understand the specific entities
-of the dataset. A sample text 
+2. Bidirectional Encoder Representations from Transformers(BERT) is a transformer-based machine learning technique for natural language processing pre-training developed by Google. We can fine-tune these BERT models on a specific task, like sentiment analysis and question answering, with our own data.
 
-```
```
Comment (sreehari-aot) [inline forms-flow-data-analysis-api/README.md:17]: {   
    "applicationId":<application id>,
    "formUrl":"<formurl>",
    "data":[{
        "text":"nice service",
        "elementId":"" (optional)
        
    }]
}

add example request body

File: forms-flow-data-analysis-api/README.md [hunk 2]
```diff
@@ -6,43 +6,32 @@
...
-        {
-            "entities": [
-                (4, 11, "SER", "pos"),
-                (15, 23, "LOC", "pos"),
-                (68, 73, "STAFF", "pos"),
-            ]
-        },
-    )
-```
+2. We will load the datasets using pandas and split into train and test sets. The input dataset was cleaned using various preprocessing techniques.
```
Comment (sreehari-aot) [inline forms-flow-data-analysis-api/README.md:13]: cleanup?

File: forms-flow-data-analysis-api/README.md [hunk 3]
```diff
@@ -6,43 +6,32 @@
...
-> text="awesome location and great staff. Staff provided excellent service."
+7. Trainer does not automatically evaluate model performance during training. You will need to pass Trainer a function to compute and report metrics. The Datasets library provides a simple accuracy function you can load with the load_metric function.
 
-```
-{
-  'sentiment': {'location': 'positive', 'facility': 'positive'},
-  'overall_sentiment': 'positive'
-}
-```
+8. Create a Trainer object with your model, training arguments, training and test datasets, and evaluation function. Then fine-tune your model by calling train(). This model reaches an accuracy of 86.
```
Comment (sreehari-aot) [inline forms-flow-data-analysis-api/README.md:35]: MODEL_ID is the env for selecting the model for the sentiment analysis component, user can fine-tune the model on their own dataset and use the model by providing the hugging face model id

File: forms-flow-data-analysis-api/README.md [hunk 4]
```diff
@@ -6,43 +6,32 @@
...
 
 ## Steps for enabling Sentiment Analysis component
```
Comment (sreehari-aot) [inline forms-flow-data-analysis-api/README.md:36]: Include Credits,

File: forms-flow-data-analysis-api/README.md [hunk 5]
```diff
@@ -6,43 +6,32 @@
...
 
-> text="awesome location and great staff. Staff provided excellent service."
+7. Trainer does not automatically evaluate model performance during training. You will need to pass Trainer a function to compute and report metrics. The Datasets library provides a simple accuracy function you can load with the load_metric function.
 
-```
-{
-  'sentiment': {'location': 'positive', 'facility': 'positive'},
-  'overall_sentiment': 'positive'
-}
-```
```
Comment (sreehari-aot) [inline forms-flow-data-analysis-api/README.md:45]: add MODEL_ID=<hugging face model id> to sample.env

**Review by sreehari-aot — CHANGES_REQUESTED** [review]

File: forms-flow-data-analysis-api/README.md [hunk 6]
```diff
@@ -1,64 +1,154 @@
...
+    "data":[{
+        "text":"{Sample text}",
+        "elementId":""
+    }]
 }
+
 ```
 
+## Model description [xaqren/sentiment_analysis](https://huggingface.co/xaqren/sentiment_analysis)
+
```
Comment (sreehari-aot) [inline forms-flow-data-analysis-api/README.md:140]: @seethalelias-aot here we can go with the generalized model that you trained, we don't need any service bc specific data in opensource, change that in env too

File: forms-flow-data-analysis-api/README.md [hunk 7]
```diff
@@ -1,64 +1,154 @@
...
-output of sentiment API in Redash dashboards. This information can be found in the **Insights section**
-for staff user formsflow.ai.
-
-A potential architectural conflict, which can happen is the choosing of
-postgres database and mongodb database. What database should we use, for
-which arhchitecure. 
+One of the unique features of the formsflow.ai framework is Sentiment Analysis. It can analyze the sentiment from forms based on specific topics mentioned by the designer during form creation.
+
+* A form designer can drag and drop **Text Area with Analytics component** and in section **Data** add key topics for Sentiment Analysis like facility, service, etc. This activates sentiment analysis component.
+* Based on the input responses of the user formsflow.ai process sentiment associated with each user's responses and stores it MongoDB database using **Python API**.
```
Comment (sreehari-aot) [inline forms-flow-data-analysis-api/README.md:150]: the response will be patched to mongo by Camunda listener

File: forms-flow-data-analysis-api/sample.env [hunk 8]
```diff
@@ -7,6 +7,7 @@ DATA_ANALYSIS_DB_PASSWORD=changeme
 DATA_ANALYSIS_DB_NAME=dataanalysis
 DATA_ANALYSIS_DB_URL=postgresql://general:changeme@forms-flow-data-analysis-db:5432/dataanalysis
 DATA_ANALYSIS_API_BASE_URL=http://localhost:5000
+MODEL_ID=xaqren/sentiment_analysis
```
Comment (sreehari-aot) [inline forms-flow-data-analysis-api/sample.env:11]: here too change to generic

**Review by sreehari-aot — CHANGES_REQUESTED** [review]

File: forms-flow-data-analysis-api/README.md (hunk 6 above)
Comment (seethalelias-aot) [reply forms-flow-data-analysis-api/README.md:140]: ok

File: forms-flow-data-analysis-api/README.md [hunk 9]
```diff
@@ -1,64 +1,154 @@
...
+`KEYCLOAK_URL_REALM`|The Keycloak realm to use|eg. forms-flow-ai | `forms-flow-ai`
+`KEYCLOAK_WEB_CLIENT_ID`|Client ID for Web to register with Keycloak|eg. forms-flow-web|`forms-flow-web`
+`KEYCLOAK_WEB_CLIENT_ID`|Client ID for Web to register with Keycloak|eg. forms-flow-web|`forms-flow-web`
+`DATA_ANALYSIS_DB_USER` |formsflow data analysis database postgres user|Used on installation to create the database.Choose your own|`general`
+`DATA_ANALYSIS_DB_PASSWORD` |formsflow data analysis database postgres password|Used on installation to create the database.Choose your own|`changeme`
+`DATA_ANALYSIS_DB_NAME` |formsflow data analysis database name|Used on installation to create the database.Choose your own|`dataanalysis`
+`DATA_ANALYSIS_DB_URL` |JDBC DB Connection URL for formsflow||`postgresql://general:changeme@forms-flow-data-analysis-db:5432/dataanalysis`
+`DATA_ANALYSIS_API_BASE_URL`:triangular_flag_on_post:|formsflow.ai Data analysis API URL||`http://{your-ip-address}:5001`
+`MODEL_ID`:triangular_flag_on_post:|formsflow.ai Hugging face model id||`xaqren/sentiment_analysis`
+
```
Comment (sreehari-aot) [inline forms-flow-data-analysis-api/README.md:65]: one more env is there
DATABASE_SUPPORT=DISABLED
possible values are DISABLED / ENABLED
forms-flow-data-analysis-api\src\api\config.py
please add that in yml file too

File: forms-flow-data-analysis-api/docker-compose-linux.yml [hunk 10]
```diff
@@ -40,6 +40,7 @@ services:
       JWT_OIDC_JWKS_CACHE_TIMEOUT: 300
       DATA_ANALYSIS_API_BASE_URL: ${DATA_ANALYSIS_API_BASE_URL}
       MODEL_ID: ${MODEL_ID:-xaqren/sentiment_analysis}
```
Comment (sreehari-aot) [inline forms-flow-data-analysis-api/docker-compose-linux.yml:42]: change model to generic

File: forms-flow-data-analysis-api/docker-compose-windows.yml [hunk 11]
```diff
@@ -41,6 +41,7 @@ services:
       JWT_OIDC_JWKS_CACHE_TIMEOUT: 300
       DATA_ANALYSIS_API_BASE_URL: ${DATA_ANALYSIS_API_BASE_URL}
       MODEL_ID: ${MODEL_ID:-xaqren/sentiment_analysis}
```
Comment (sreehari-aot) [inline forms-flow-data-analysis-api/docker-compose-windows.yml:43]: change model to generic

**Review by sreehari-aot — APPROVED** [review]
LGTM

**Review by abilpraju-aot — APPROVED** [review]

**Pull request merged**